Update app-state.md (#28339)
dotnet/AspNetCore.Docs: aspnetcore/fundamentals/app-state.md

To cache application wide data, see <xref:performance/caching/memory>.

## Checking session state

> [ISession.IsAvailable](xref:Microsoft.AspNetCore.Http.ISession.IsAvailable) is intended to check for transient failures. Calling `IsAvailable` before session is set can throw an `InvalidOperationException`.

Reviewer: ```suggestion
[ISession.IsAvailable](xref:Microsoft.AspNetCore.Http.ISession.IsAvailable) is intended to check for transient failures. Calling `IsAvailable` before the session middleware runs will throw an `InvalidOperationException`.
```

> [ISession.IsAvailable](xref:Microsoft.AspNetCore.Http.ISession.IsAvailable) is intended to check for transient failures. Calling `IsAvailable` before the session middleware runs will throw an `InvalidOperationException`.

Author: ```suggestion
[ISession.IsAvailable](xref:Microsoft.AspNetCore.Http.ISession.IsAvailable) is intended to check for transient failures. Calling `IsAvailable` before the session middleware runs throws an `InvalidOperationException`.
```
